### PR TITLE
Remove unnecessary foreign keys from gift_codes table.

### PR DESCRIPTION
**Pull request by christian-oudard**
.

### CODE REVIEW

**Review by david1alvarez — APPROVED** [review]
LGTM -- possible dupe on foreign key validations, left comment

File: full-service/src/db/gift_code.rs [hunk 1]
```diff
@@ -44,13 +44,7 @@ pub trait GiftCodeModel {
     #[allow(clippy::too_many_arguments)]
     fn create(
         gift_code_b58: &EncodedGiftCode,
-        root_entropy: Option<&RootEntropy>,
-        bip39_entropy: Option<&Vec<u8>>,
-        txo_public_key: &CompressedRistrettoPublic,
         value: i64,
-        memo: &str,
```
Comment (Shramp) [inline full-service/src/db/gift_code.rs:51]: do we need to update the documentation for this trait now that these args have been removed? looks like the doc still says:
```
    /// Gift code includes:
    /// * entropy
    /// * txo public key
    /// * memo
```

**Review by BrianCorbin — APPROVED** [review]

File: full-service/src/db/gift_code.rs (hunk 1 above)
Comment (Shramp) [reply full-service/src/db/gift_code.rs:51]: nvm. i get that those things are still there, just encoded in the b58

File: full-service/src/bin/main.rs [hunk 2]
```diff
@@ -83,15 +83,14 @@ fn main() {
...
+    // until the user resolves it.
+    WalletDb::validate_foreign_keys(&conn);
     conn.batch_execute("PRAGMA foreign_keys = OFF;")
         .expect("failed disabling foreign keys");
     embedded_migrations::run_with_output(&conn, &mut std::io::stdout())
         .expect("failed running migrations");
-
-    // Ensure the database is still in good shape. If not, we will abort until the
-    // user resolves it.
     WalletDb::validate_foreign_keys(&conn);
```
[the remark this answers is not in the document]
Comment (christian-oudard) [reply full-service/src/bin/main.rs:93]: Yes, to be sure that we do not run migrations while there are foreign key errors, and also that the migrations do not introduce foreign key errors.

**Pull request merged**